Spacy

aborsu opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
python.msgpack-numpy: init 0.4.1

Commit:
python.murmurhash: 0.26.4 -> 0.28.0

Commit:
python.thinc: 6.5.1 -> 6.10.2

aborsu commented:
I still need to add the update to spacy. 
This is to fix python.thinc which is broken on master.

hedning commented:
You need to add yourself to `maintainers/maintainer-list.nix` so that `lib.maintainers.aborsu` is defined (that's what ofborg is complaining about).

Commit:
add aborsu as maintainer

Commit:
python.ftfy: 5.3.0 -> 4.4.3

V5 only supports python3. Since at the moment the only packages
that use ftfy are spacy and textacy which both support
python2 and 3, I propose to roll back to v4 until another package
requires v5, at that point we can make a duplicate package.

Commit:
python.spacy: 1.8.2 -> 2.0.9

aborsu commented:
Fix thinc and spacy for #36453

dotlambda commented on pkgs/development/python-modules/spacy/default.nix:
```diff
@@ -3,53 +3,68 @@
+    sha256 = "1ihkhflhyz67bp73kfjqfrbcgdxi2msz5asbrh0pkk590c4vmms5";
   };
 
+  preBuild = ''
```
`prePatch`

dotlambda commented on pkgs/development/python-modules/thinc/default.nix:
```diff
@@ -64,22 +53,34 @@ in buildPythonPackage rec {
+    pytest
   ];
 
+  preBuild = ''
```
`prePatch`

dotlambda commented on pkgs/development/python-modules/ftfy/default.nix:
```diff
@@ -7,18 +7,17 @@
 buildPythonPackage rec {
-  name = "${pname}-${version}";
   pname = "ftfy";
-  # latest is 5.1.1, buy spaCy requires 4.4.3
```
Why did you drop the comment?
Please open a PR at the spaCy side that allows it to use a newer ftfy version.

aborsu replied:
I can open a PR but I very much doubt that it will be accepted.
Spacy is still very much supporting python2.7 and ftfy v5 dropped support for python3.
The v4 branch of ftfy still gets some updates so I doubt the maintainer of Spacy will accept working with two different versions of ftfy as that would make keeping the 2.7 compatibility that much harder.

dotlambda replied:
As long as nobody needs a newer version of ftfy, I think it's okay to stay at the old one. However, please keep the comment.

dotlambda commented on pkgs/development/python-modules/msgpack-numpy/default.nix:
```diff
@@ -0,0 +1,40 @@
+   numpy
+  ];
+
+  doCheck = true;
```
not needed

aborsu replied:
Why not? There are tests, they work an they take less than 10ms to run?

dotlambda replied:
`doCheck` is set to true by default

aborsu replied:
Ah... indeed

aborsu commented:
@dotlambda I have made the requested changes.

dotlambda commented:
@GrahamcOfBorg build python2.pkgs.msgpack-numpy python3.pkgs.msgpack-numpy python2.pkgs.murmurhash python3.pkgs.murmurhash python2.pkgs.thinc python3.pkgs.thinc python2.pkgs.ftfy python3.pkgs.ftfy python2.pkgs.spacy python3.pkgs.spacy

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36899&attempt_id=266c348d-0f62-45a2-a884-32eeff02921d)

Attempted: python2.pkgs.msgpack-numpy, python3.pkgs.msgpack-numpy, python2.pkgs.murmurhash, python3.pkgs.murmurhash, python2.pkgs.thinc, python3.pkgs.thinc, python2.pkgs.ftfy, python3.pkgs.ftfy, python2.pkgs.spacy, python3.pkgs.spacy

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/8p2frarb887kbd2l2pqw7vkfkc691r69-python2.7-msgpack-numpy-0.4.1
/nix/store/60k4zv5k8wpfw84chsjmifh8dp7la2vm-python3.6-msgpack-numpy-0.4.1
/nix/store/kg4kssf85a045pj4fp00skdavr0cdd91-python2.7-murmurhash-0.28.0
/nix/store/hi14f7n56c6w1jlx0s9jzwgav6kf63id-python3.6-murmurhash-0.28.0
/nix/store/x6dzr8vmw8j94faqny1i945kb7v37ij6-python2.7-thinc-6.10.2
/nix/store/nv8fbgj4315pyqkn009z4jhwn3raia3j-python3.6-thinc-6.10.2
/nix/store/gx23rc5iwzsrv9142xizzcayg8jjj2vz-python2.7-ftfy-4.4.3
/nix/store/0n6jhisnavr9xxm1ghv8lkipp7cf6icr-python3.6-ftfy-4.4.3
/nix/store/wwdrzvgygpgxahf1rr5qyxjlb328phx2-python2.7-spacy-2.0.9
/nix/store/f2igd9wip42jqazjssni8d2hwfl6pwxv-python3.6-spacy-2.0.9
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36899&attempt_id=16d73023-41a8-493b-b869-723b3193b0f4)

Attempted: python2.pkgs.msgpack-numpy, python3.pkgs.msgpack-numpy, python2.pkgs.murmurhash, python3.pkgs.murmurhash, python2.pkgs.thinc, python3.pkgs.thinc, python2.pkgs.ftfy, python3.pkgs.ftfy, python2.pkgs.spacy, python3.pkgs.spacy

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/a8m0jxc94qnql0g0i5lgdqwv8hf9k0cz-python2.7-msgpack-numpy-0.4.1
/nix/store/41c5qafq4avpn59sp1asfnpy3bxc9dbf-python3.6-msgpack-numpy-0.4.1
/nix/store/5z5jbhimgkghl81fbl1f41yg2c15s47r-python2.7-murmurhash-0.28.0
/nix/store/rlc4cg5cn4afa3bw2qcd7l80qg23mbf1-python3.6-murmurhash-0.28.0
/nix/store/ri3mmr3xjw3jni3j1jl6v6brydmmf6in-python2.7-thinc-6.10.2
/nix/store/ffhrzipq8v4z38br4j80j84c6lg6h2aj-python3.6-thinc-6.10.2
/nix/store/kljyqkng0jihgrhgzq58m04pvq8j1l0r-python2.7-ftfy-4.4.3
/nix/store/d4dkmzd85appmvk9rqq3qplpsxwkkw6f-python3.6-ftfy-4.4.3
/nix/store/if81f48mh9zf95ixvz2yl7k6q4d5rqjw-python2.7-spacy-2.0.9
/nix/store/ldin5b4dj6j1d234a4rgzdxji9vnrhfx-python3.6-spacy-2.0.9
```
</p></details>

dotlambda commented:
@aborsu Btw, ftfy will probably be upgraded again with the next automatic package update. Thinking about it, we probably should just have disabled spacy for Python 3. Maybe you can find out if spacy works with ftfy v5 in the meantime. At least, it will work fine for 18.03 now.

aborsu commented:
@dotlambda I would be saddened if spacy was disabled on python3 as I personally don't use python2 anymore.

dotlambda commented:
Oh sorry, it's the other way round. Ftfy v5 only works with Python 3.
`pythonPackages` should be a consistent set of packages. Therefore, we usually do not want multiple versions of the same package in there.
